external-crates/move/crates/move-vm-profiler/src/lib.rs

```diff
@@ -184,23 +174,24 @@ impl GasProfiler {
     }
 
     pub fn to_file(&self) {
-        if !*PROFILER_ENABLED || !self.is_metered() {
+        let config;
+        if let Some(c) = &self.config {
+            config = c
+        } else {
             return;
         }
-        // Get the unix timestamp
-        let now = std::time::SystemTime::now()
-            .duration_since(std::time::UNIX_EPOCH)
-            .expect("Clock may have gone backwards")
-            .as_nanos();
-
-        let mut p = self.config.base_path.clone();
-        p.push(format!("gas_profile_{}_{}.json", self.profile_name(), now));
+        if !self.is_metered() {
```

Comment:
Nit: a perfect use-case for let-else!
```suggestion
        let Some(config) = &self.config else {
            return;
        };
        if !self.is_metered() {
```